confidant/routes/services.py

```diff
@@ -72,41 +72,33 @@ def get_service(id):
         'update': False,
     }
     metadata_only = request.args.get('metadata_only', default=False, type=bool)
-    if authnz.user_is_user_type('service'):
-        if not authnz.user_is_service(id):
-            logging.warning('Authz failed for service {0}.'.format(id))
-            msg = 'Service is not authorized.'
-            return jsonify({'error': msg}), 401
-        permissions['metadata'] = True
-        permissions['get'] = True
-    else:
-        logged_in_user = authnz.get_logged_in_user()
-        action = 'metadata' if metadata_only else 'get'
-        permissions['metadata'] = acl_module_check(
-            resource_type='service',
-            action=action,
-            resource_id=id,
-        )
-        permissions['get'] = acl_module_check(
-            resource_type='service',
-            action=action,
-            resource_id=id,
+    logged_in_user = authnz.get_logged_in_user()
```

Comment:
having trouble seeing it at moment, but if it's a service, will this return the service name?

Reply:
nevermind, i think i see it, it will return the service/role name

https://github.com/lyft/confidant/blob/master/confidant/authnz/__init__.py#L191

Reply:
Yep! We set g.username whether it's user or service auth.